src/transformers/modeling_utils.py

```diff
@@ -4120,15 +4120,38 @@ def _load_pretrained_model(
         expected_keys = list(model_state_dict.keys())
         prefix = model.base_model_prefix
 
+        error_msgs = []
+
+        old_keys = []
+        new_keys = []
+        renamed_keys = {}
+
+        warning_msg = f"This model {type(model)}"
+
+        original_loaded_keys = loaded_keys
+
         def _fix_key(key):
             if "beta" in key:
                 return key.replace("beta", "bias")
             if "gamma" in key:
                 return key.replace("gamma", "weight")
-            return key
-
-        original_loaded_keys = loaded_keys
-        loaded_keys = [_fix_key(key) for key in loaded_keys]
+            return None
+
+        for i, key in enumerate(loaded_keys):
+            new_key = _fix_key(key)
+            if new_key:
+                old_keys.append(key)
+                new_keys.append(new_key)
+                renamed_keys[key] = new_key
+                loaded_keys[i] = new_key
```

Comment:
This doesn't work - this will modify both `loaded_keys` and `original_loaded_keys`: 

```py
In [1]: li_0 = list(range(10))

In [2]: li_1 = li_0

In [3]: for i in range(10):
   ...:     if i % 2:
   ...:         li_0[i] = -1
   ...:

In [4]: li_0
Out[4]: [0, -1, 2, -1, 4, -1, 6, -1, 8, -1]

In [5]: li_1
Out[5]: [0, -1, 2, -1, 4, -1, 6, -1, 8, -1]
```

Reply:
Thank you sir ! You are totally right , I forget add a copy() to ensure that the original_loaded_keys remain unaltered.